src/embed.ts

```diff
@@ -304,7 +304,7 @@ async function _embed(
 
   const mode = guessMode(spec, opts.mode);
 
-  let vgSpec: VgSpec = PREPROCESSOR[mode](spec, config);
+  let vgSpec: VgSpec = PREPROCESSOR[mode](spec, { ...config, logger });
```

Comment:
Can be added directly to the passed config, since the whole arg will be discarded when the mode is not 'vega-lite'